### DISCUSSION TITLE
Multimedia Tooling

### DISCUSSION THREAD

**Discussion by BradKML** [SmartManoj/Kevin#146] [General]
When I first saw this one I got a little bit inspired for the amount of support for vast diversity of media https://github.com/rmusser01/tldw
But not just from the document reading or web crawling side, but multimedia handling side of things as well https://github.com/SmartManoj/Kevin/discussions/138
Podcasts, live streams, presentations, video essays, theater transcriptions, documentaries... Most of these use something like OpenAI's Whisper model for voice, maybe video VLMs for visuals, but even then there are options

Major hurdles:
1. Multiple speakers, OR speaker vs video clippings => https://en.wikipedia.org/wiki/Speaker_diarisation
2. Summarization and relying on tools like "Chain of Density" => https://www.prompthub.us/blog/better-summarization-with-chain-of-density-prompting
3. Chunking of information on multiple layers (document-based chunking first then recursive and semantic chunking) while handling them with RAG => https://medium.com/@anuragmishra_27746/five-levels-of-chunking-strategies-in-rag-notes-from-gregs-video-7b735895694d
4. Pure audio transcriptions vs inclusion on what is "in frame" for video => see @byjlw for example
5. Multi-lingual support for media from other countries => BLOOM and other multi-lingual LLMs, requires advancements in multi-lingual Whisper
6. Eliminating music to prevent accidental mis-encoding => vocal isolation or instrumental isolation (commonly used in music remixing and sampling)

Examples:
- https://github.com/rmusser01/tldw
- https://github.com/byjlw/video-analyzer
- https://github.com/martinopiaggi/summarize
- https://github.com/DevRico003/youtube_summarizer
- https://github.com/inboxpraveen/LLM-Minutes-of-Meeting
- https://github.com/sidedwards/ai-video-summarizer
- https://github.com/mickekring/ragnar
- https://github.com/video-db/Director
- https://github.com/jinhoyoo/summarize_youtube_video_lecture
- https://github.com/thewh1teagle/vibe

**Comment by SmartManoj** [comment]
Would you happen to have any video with the expected output to test?

**Reply by BradKML** [reply]
With all these tools I can imagine this pipeline:
1. Provide a list of video/audio, or a channel, or even a list of channels, to download
2. For each piece of video/audio, create general transcript with high accuracy first
3. Separate main speaker(s), guest speaker(s), music (e.g. background, show intro), and media clippings (e.g. news, speech reference)
4. Focus on transcribing and tagging the main speaker(s) and guest speaker(s), with media clippings treated as quotations
5. RAG them based on 4-8 minute segments AND on episode-level or video-level (with monologue/response vs dialogue awareness)
6. Create CoD summaries for episode-level or video level media chunks with context on who says/reports what

Bonus: the pipeline can also recognize what is on video when things are happening (e.g. live news reporting), and can extract accurate thumbnails, and this would be stored as the segment-level.
